Userland/Libraries/LibGL/SoftwareGLContext.cpp

```diff
@@ -3009,4 +3006,173 @@ void SoftwareGLContext::sync_device_texcoord_config()
+            material.specular = { params[0], params[1], params[2], params[2] };
+        } else if (pname == GL_EMISSION) {
+            material.emissive = { params[0], params[1], params[2], params[2] };
+        } else if (pname == GL_SHININESS) {
+            material.shininess = *params;
+        } else if (pname == GL_AMBIENT_AND_DIFFUSE) {
+            material.ambient = { params[0], params[1], params[2], params[2] };
+            material.diffuse = { params[0], params[1], params[2], params[2] };
+        }
+        break;
+    }
+    case GL_BACK: {
+        auto& material = m_rasterizer.material(1);
+        if (pname == GL_AMBIENT) {
+            material.ambient = { params[0], params[1], params[2], params[2] };
+        } else if (pname == GL_DIFFUSE) {
+            material.ambient = { params[0], params[1], params[2], params[2] };
+        } else if (pname == GL_SPECULAR) {
+            material.specular = { params[0], params[1], params[2], params[2] };
+        } else if (pname == GL_EMISSION) {
+            material.emissive = { params[0], params[1], params[2], params[2] };
+        } else if (pname == GL_SHININESS) {
+            material.shininess = *params;
+        } else if (pname == GL_AMBIENT_AND_DIFFUSE) {
+            material.ambient = { params[0], params[1], params[2], params[2] };
+            material.diffuse = { params[0], params[1], params[2], params[2] };
+        }
+        break;
+    }
+    case GL_FRONT_AND_BACK: {
+        auto& front_material = m_rasterizer.material(0);
+        auto& back_material = m_rasterizer.material(1);
+        if (pname == GL_AMBIENT) {
+            front_material.ambient = { params[0], params[1], params[2], params[2] };
+            back_material.ambient = { params[0], params[1], params[2], params[2] };
+        } else if (pname == GL_DIFFUSE) {
+            front_material.ambient = { params[0], params[1], params[2], params[2] };
+            back_material.ambient = { params[0], params[1], params[2], params[2] };
+        } else if (pname == GL_SPECULAR) {
+            front_material.specular = { params[0], params[1], params[2], params[2] };
+            back_material.specular = { params[0], params[1], params[2], params[2] };
+        } else if (pname == GL_EMISSION) {
+            front_material.emissive = { params[0], params[1], params[2], params[2] };
+            back_material.emissive = { params[0], params[1], params[2], params[2] };
+        } else if (pname == GL_SHININESS) {
+            front_material.shininess = *params;
+            back_material.shininess = *params;
+        } else if (pname == GL_AMBIENT_AND_DIFFUSE) {
+            front_material.ambient = { params[0], params[1], params[2], params[2] };
+            front_material.diffuse = { params[0], params[1], params[2], params[2] };
+            back_material.ambient = { params[0], params[1], params[2], params[2] };
+            back_material.diffuse = { params[0], params[1], params[2], params[2] };
+        }
```

Comment:
We can reduce this code quite a bit if you add a lambda and then invoke that on `.material(0)` and `.material(1)` based on `face`'s value